File: frontend/src/types/prediction.ts
export type Furnishing =
  | "Furnished"
  | "Semi-Furnished"
  | "Unfurnished";

export type Transaction =
  | "New Property"
  | "Resale"
  | "Other"
  | "Rent/Lease";

export type Ownership =
  | "Freehold"
  | "Co-operative Society"
  | "Power Of Attorney"
  | "Leasehold";

export type Facing =
  | "East"
  | "West"
  | "North"
  | "South"
  | "North - East"
  | "North - West"
  | "South - East"
  | "South -West";

export interface PredictionRequest {
  location: string;
  carpet_area_sqft: number;
  floor_num: number;
  bathroom: number;
  balcony: number;
  car_parking: number;
  furnishing: Furnishing;
  transaction: Transaction;
  ownership: Ownership;
  facing: Facing;
}

export interface PredictionResponse {
  predicted_price: number;
}

export interface PredictionResultState {
  prediction: number;
  property: PredictionRequest;
}

File: backend/app/schemas/prediction.py
from typing import Literal

from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    location: str = Field(min_length=1)
    carpet_area_sqft: float = Field(gt=0)
    floor_num: int = Field(ge=-2)
    bathroom: int = Field(ge=0, le=20)
    balcony: int = Field(ge=0, le=20)
    car_parking: int = Field(default=0, ge=0)
    furnishing: Literal["Furnished", "Semi-Furnished", "Unfurnished"]
    transaction: Literal["New Property", "Resale", "Other", "Rent/Lease"]
    ownership: Literal[
        "Freehold",
        "Co-operative Society",
        "Power Of Attorney",
        "Leasehold",
    ]
    facing: Literal[
        "East",
        "West",
        "North",
        "South",
        "North - East",
        "North - West",
        "South - East",
        "South -West",
    ]


class PredictionResponse(BaseModel):
    predicted_price: float
